### PR TITLE
feat: implement sending transaction through wallet connect

### LINKED ISSUE #124: ️🛠 - Implement sending transaction event
### Description

If we receive the eth_sendTransaction or eth_sendRawTransaction event, we should open the TransactionSummary popup, as well as the DApp that sent the event. On Confirm, the transaction should be sent to the specific network, as well as the popup closed.

### Requirements

- [ ]  Open `TransactionSummaryPopup` for transaction
- [ ]  On confirm, send the transaction

### Checklist

- [X] I have linked the correct milestone and/or project(s)
- [X] I have added the appropriate labels
- [x] I have assigned this task to the correct people

### PR DESCRIPTION
**Pull request by Tuditi**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/components/popup/popups/SignEvmTransactionPopup.svelte
Comment (MarkNerdi) [inline packages/desktop/components/popup/popups/SignEvmTransactionPopup.svelte:1]: Could you rename this file + corresponding `PopupId` to something more general e.g. `DappEvmTransactionPopup`

File: packages/desktop/components/popup/popups/SignEvmTransactionPopup.svelte [hunk 1]
```diff
@@ -81,11 +84,16 @@
     async function sign(): Promise<void> {
         isBusy = true
         try {
-            const result = await signEvmTransaction(transaction, chainId, account)
+            let result: string | TransactionReceipt = await signEvmTransaction(transaction, chainId, account)
+
+            if (signAndSend) {
+                const provider = chain.getProvider()
+                result = await provider?.eth.sendSignedTransaction(result)
+            }
```
Comment (MarkNerdi) [inline packages/desktop/components/popup/popups/SignEvmTransactionPopup.svelte]: could you reuse the `signAndSendEvmTransaction` , maybe with a flag for this here? it includes the same logic + sets the loading state

Comment (MarkNerdi) [reply packages/desktop/components/popup/popups/SignEvmTransactionPopup.svelte]: as well we'd need to include the rest of the code you can see in `sendTransactionFromEvm` e.g. persistingEvmTransactions

File: packages/shared/src/lib/core/wallet/actions/signEvmTransaction.ts [hunk 2]
```diff
@@ -21,14 +20,15 @@ export async function signEvmTransaction(
     }
     const { index } = account
 
+    const transactionCopy = { ...transaction }
```
Comment (MarkNerdi) [inline packages/shared/src/lib/core/wallet/actions/signEvmTransaction.ts]: pls use `structuredClone`. With that we're 100% sure it is a deep clone

File: packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts [hunk 3]
```diff
@@ -15,26 +15,29 @@ import {
...
-                provider,
-                account
-            )
+            const signedTransaction = await signEvmTransaction(preparedTransaction, chain, account)
+
+            if (!signAndSend || !signedTransaction) {
+                if (callback && typeof callback === 'function') {
+                    callback({ result: signedTransaction })
+                }
+                return signedTransaction
```
Comment (MarkNerdi) [inline packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts]: we just need `return` as we expect a  `Promise<void>`

File: packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts [hunk 4]
```diff
@@ -15,26 +15,29 @@ import {
...
-                preparedTransaction,
-                chainId,
-                coinType,
-                provider,
-                account
-            )
+            const signedTransaction = await signEvmTransaction(preparedTransaction, chain, account)
+
+            if (!signAndSend || !signedTransaction) {
+                if (callback && typeof callback === 'function') {
```
Comment (MarkNerdi) [inline packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts]: i think here and line `54` we dont need the ` && typeof callback === 'function'` part,  as thats already taken care of with TS

File: packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts [hunk 5]
```diff
@@ -15,26 +15,29 @@ import {
...
+                    callback({ result: signedTransaction })
+                }
+                return signedTransaction
+            }
+
+            const transactionReceipt = await sendSignedEvmTransaction(chain, signedTransaction)
             if (!transactionReceipt) {
                 return
             }
```
Comment (MarkNerdi) [inline packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts]: do we also need the `callback` here?

**Review by MarkNerdi — CHANGES_REQUESTED** [review]

File: packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts (hunk 4 above)
Comment (Tuditi) [reply packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts]: It's useful for runtime checking, but I checked and we specify the callback each time, so I applied your suggestion.

File: packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts (hunk 5 above)
Comment (Tuditi) [reply packages/shared/src/lib/core/wallet/actions/send/sendTransactionFromEvm.ts]: Yup, I added it.

**Review by MarkNerdi — APPROVED** [review]

**Pull request merged**